server/cmd/sitemap.go

```diff
@@ -0,0 +1,103 @@
+// Copyright © 2018 NAME HERE <EMAIL ADDRESS>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/Coderockr/vitrine-social/server/db"
+	"github.com/Coderockr/vitrine-social/server/db/repo"
+	"github.com/Coderockr/vitrine-social/server/model"
+	"github.com/spf13/cobra"
+	"github.com/thecodenation/go-sitemap-generator/stm"
+)
+
+// sitemapCmd represents the sitemap command
+var sitemapCmd = &cobra.Command{
+	Aliases: []string{"sitemap"},
+	Use:     "sitemap-generate",
+	Short:   "Generate sitemap",
+	Run:     withEnvironment(sitemapFunc),
+}
+
+func init() {
+	rootCmd.AddCommand(sitemapCmd)
+}
+
+func sitemapFunc(cmd *cobra.Command, args []string) {
+	var o []*model.Organization
+	sm := generateSitemap()
+
+	conn, err := db.GetFromEnv()
+	if err != nil {
+		log.Fatal(err.Error())
+		os.Exit(1)
+	}
+
+	rO := repo.NewOrganizationRepository(conn)
+	rN := repo.NewNeedRepository(conn)
+	o, err = rO.GetAll()
+	if err != nil {
+		log.Fatal(err.Error())
+		os.Exit(1)
+	}
+	for _, v := range o {
+		sm.Add(stm.URL{"loc": fmt.Sprintf("entidade/%d", v.ID), "changefreq": "hourly", "priority": 0.8})
+		n, err := rN.GetOrganizationNeeds(v.ID, "id", "asc")
+		if err != nil {
+			log.Fatal(err.Error())
+			os.Exit(1)
+		}
+		for _, k := range n {
+			sm.Add(stm.URL{"loc": fmt.Sprintf("%s/detalhes/%d", os.Getenv("FRONTEND_URL"), k.ID), "changefreq": "hourly", "priority": 0.8})
```

Comment:
não seria melhor usar um formato de "notícia"? isso não ajudaria na indexação?

https://support.google.com/news/publisher-center/answer/74288?hl=pt-BR

Reply:
Não cheguei a usar esse cara. Na Code:Nation usamos esse formato e o pessoal de marketing aprovou :) A única diferença é que nós usamos slugs como:

http://www.codenation.com.br/journey/frontend/challenge/react-0.html

E não os ids. Mas não sei até onde isso faz diferença neste caso

Reply:
humm... soh que vcs estão usando JAM Stack ai, quando entra nesse "http://www.codenation.com.br/journey/frontend/challenge/react-0.html" tem o conteúdo de verdade... no nosso é soh aquele load do react "https://vitrinesocial.org/detalhes/122" (da um curl para ver)... acho nesse caso, se não formos adicionar metadados aqui, seria melhor usar a URL do `shareNeed` (https://api.vitrinesocial.org/v1/need/${needID}/share) que dai vai ter algo para ele olhar